src/php/datetime/strtotime.js

```diff
@@ -1,300 +1,1144 @@
+    x: 9,
+    nov: 10,
+    november: 10,
+    xi: 10,
+    dec: 11,
+    december: 11,
+    xii: 11
+  }[monthStr.toLowerCase()]
+}
+
+function lookupWeekday (dayStr, desiredSundayNumber = 0) {
+  const dayNumbers = {
+    mon: 1,
+    monday: 1,
+    tue: 2,
+    tuesday: 2,
+    wed: 3,
+    wednesday: 3,
+    thu: 4,
+    thursday: 4,
+    fri: 5,
+    friday: 5,
+    sat: 6,
+    saturday: 6,
+    sun: 0,
+    sunday: 0
   }
-  if (!isNaN(parsed = Date.parse(text))) {
-    return parsed / 1000 | 0
+
+  return dayNumbers[dayStr.toLowerCase()] || desiredSundayNumber
+}
+
+function lookupRelative (relText) {
+  const relativeNumbers = {
+    last: -1,
+    previous: -1,
+    this: 0,
+    first: 1,
+    next: 1,
+    second: 2,
+    third: 3,
+    fourth: 4,
+    fifth: 5,
+    sixth: 6,
+    seventh: 7,
+    eight: 8,
+    eighth: 8,
+    ninth: 9,
+    tenth: 10,
+    eleventh: 11,
+    twelfth: 12
   }
-  // Browsers !== Chrome have problems parsing ISO 8601 date strings, as they do
-  // not accept lower case characters, space, or shortened time zones.
-  // Therefore, fix these problems and try again.
-  // Examples:
-  //   2015-04-15 20:33:59+02
-  //   2015-04-15 20:33:59z
-  //   2015-04-15t20:33:59+02:00
-  pattern = new RegExp([
-    '^([0-9]{4}-[0-9]{2}-[0-9]{2})',
-    '[ t]',
-    '([0-9]{2}:[0-9]{2}:[0-9]{2}(\\.[0-9]+)?)',
-    '([\\+-][0-9]{2}(:[0-9]{2})?|z)'
-  ].join(''))
-  match = text.match(pattern)
-  if (match) {
-    // @todo: time zone information
-    if (match[4] === 'z') {
-      match[4] = 'Z'
-    } else if (match[4].match(/^([+-][0-9]{2})$/)) {
-      match[4] = match[4] + ':00'
-    }
-
-    if (!isNaN(parsed = Date.parse(match[1] + 'T' + match[2] + match[4]))) {
-      return parsed / 1000 | 0
-    }
+
+  const relativeBehavior = {
+    this: 1
+  }
+
+  const relTextLower = relText.toLowerCase()
+
+  return {
+    amount: relativeNumbers[relTextLower],
+    behavior: relativeBehavior[relTextLower] || 0
   }
+}
 
-  date = now ? new Date(now * 1000) : new Date()
-  days = {
-    'sun': 0,
-    'mon': 1,
-    'tue': 2,
-    'wed': 3,
-    'thu': 4,
-    'fri': 5,
-    'sat': 6
+function processTzCorrection (tzOffset, oldValue) {
+  const reTzCorrectionLoose = /(?:GMT)?([+-])(\d+)(:?)(\d{0,2})/i
+  tzOffset = tzOffset && tzOffset.match(reTzCorrectionLoose)
+
+  if (!tzOffset) {
+    return oldValue
   }
-  ranges = {
-    'yea': 'FullYear',
-    'mon': 'Month',
-    'day': 'Date',
-    'hou': 'Hours',
-    'min': 'Minutes',
-    'sec': 'Seconds'
+
+  let sign = tzOffset[1] === '-' ? 1 : -1
+  let hours = +tzOffset[2]
+  let minutes = +tzOffset[4]
+
+  if (!tzOffset[4] && !tzOffset[3]) {
+    minutes = Math.floor(hours % 100)
+    hours = Math.floor(hours / 100)
   }
 
-  function lastNext (type, range, modifier) {
-    var diff
-    var day = days[range]
+  return sign * (hours * 60 + minutes)
+}
+
+const formats = {
+  yesterday: {
+    regex: /^yesterday/i,
+    name: 'yesterday',
+    callback: function () {
```

Comment:
(Not a big deal to me, but just for a quick friendly review, I might add a few comments as this...)

As long as you aren't using the function as a class (in which case you should probably be using an ES6 class anyways), the shorthand function method is somewhat nicer to go with...i.e., replace this line with just:

```js
callback () {
```

Reply:
I think I will follow your suggestion, because at one point I was thinking about the same. I just wasn't sure how much of the "new" JS syntax I'm allowed to use in locutus.

Reply:
ES6 is okay!